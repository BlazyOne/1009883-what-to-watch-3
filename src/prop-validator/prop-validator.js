import PropTypes from 'prop-types';

const PropValidator = {
  PROMO_FILM: PropTypes.exact({
    title: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    releaseDate: PropTypes.string.isRequired
  }),
  FILMS: PropTypes.arrayOf(
      PropTypes.exact({
        id: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
        image: PropTypes.string.isRequired
      })
  ),
  FILM: PropTypes.exact({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  }),
  FILM_INFO: PropTypes.exact({
    id: PropTypes.string.isRequired,
    backgroundImage: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    year: PropTypes.string.isRequired,
    poster: PropTypes.string.isRequired,
    ratingScore: PropTypes.string.isRequired,
    ratingLevel: PropTypes.string.isRequired,
    ratingCount: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    director: PropTypes.string.isRequired,
    starring: PropTypes.string.isRequired
  }),
  ON_TITLE_CLICK: PropTypes.func.isRequired,
  ON_CARD_CLICK: PropTypes.func.isRequired,
  ON_MOUSE_OVER_CARD: PropTypes.func.isRequired
};

export {PropValidator};
