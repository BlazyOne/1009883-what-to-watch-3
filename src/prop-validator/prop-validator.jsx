import PropTypes from 'prop-types';

const PropValidator = {
  PROMO_FILM: PropTypes.shape({
    title: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    releaseDate: PropTypes.string.isRequired
  }),
  FILMS: PropTypes.arrayOf(PropTypes.string.isRequired),
  ON_TITLE_CLICK: PropTypes.func.isRequired
};

export {PropValidator};
