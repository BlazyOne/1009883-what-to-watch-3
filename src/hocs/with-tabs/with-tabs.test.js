import React from 'react';
import renderer from 'react-test-renderer';
import withTabs from './with-tabs.jsx';
import {PropValidator} from '../../prop-validator/prop-validator.js';

const MockComponent = (props) => {
  const {currentTab, onTabChange} = props;

  return (
    <ul>
      <li
        className={currentTab === `overview` || currentTab === `start` ?
          `active` :
          ``}
        onClick={() => {
          onTabChange(`overview`);
        }}
      >
        Overview
      </li>
      <li
        className={currentTab === `details` ?
          `active` :
          ``}
        onClick={() => {
          onTabChange(`details`);
        }}
      >
        Details
      </li>
    </ul>
  );
};

MockComponent.propTypes = {
  currentTab: PropValidator.CURRENT_TAB,
  onTabChange: PropValidator.ON_TAB_CHANGE
};

const MockComponentWrapped = withTabs(MockComponent);

it(`Should withTabs render correctly`, () => {
  const tree = renderer.create(
      <MockComponentWrapped/>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
