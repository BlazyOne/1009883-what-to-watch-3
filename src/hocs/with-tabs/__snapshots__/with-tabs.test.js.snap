// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should withTabs render correctly 1`] = `
<ul>
  <li
    className="active"
    onClick={[Function]}
  >
    Overview
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    Details
  </li>
</ul>
`;
